### PR TITLE
feat: support filter pattern expressions for CIS benchmark regarding alarms

### PR DESCRIPTION
**Pull request by rogeriopeixotocx**
**Proposed Changes**
- [x] Define grammar to support [AWS CloudWatch filter and patterns expression syntax](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html)
- [x] Generate parser and implement visitors
- [x] Implement error handler
- [x] Update CI pipelines to support ANTLR v4
- [x] Update Dockerfiles to support ANTLR v4
- [x] Update Makefile
- [x] Unit tests for JSONFIlter parser
- [x] Detect when JSONFilter Parser is needed in resolvers @joaoReigota1 
- [x] Inject parsed structure into KICS payload @joaoReigota1 
- [x] Add example rego query (4.1 rule)

**CIS Alarm Rules**
- [x] 4.1 - Ensure a log metric filter and alarm exist for unauthorized API calls @rogeriopeixotocx 
- [x] 4.2 - Ensure a log metric filter and alarm exist for Management Console sign-in without MFA @joaoReigota1 
- [x] 4.3 - Ensure a log metric filter and alarm exist for usage of 'root' account @rogeriopeixotocx 
- [x] 4.4 - Ensure a log metric filter and alarm exist for IAM policy changes @felipe-avelar 
- [x] 4.5 - Ensure a log metric filter and alarm exist for CloudTrail configuration changes @joaoReigota1 
- [x] 4.6 - Ensure a log metric filter and alarm exist for AWS Management Console authentication failures @rogeriopeixotocx 
- [x] 4.7 - Ensure a log metric filter and alarm exist for disabling or scheduled deletion of customer-created CMKs
- [x] 4.8 - Ensure a log metric filter and alarm exist for S3 bucket policy changes @rogeriopeixotocx 
- [x] 4.9 - Ensure a log metric filter and alarm exist for AWS Config configuration changes
- [x] 4.10 - Ensure a log metric filter and alarm exist for security group changes
- [x] 4.11 - Ensure a log metric filter and alarm exist for changes to NACL
- [x] 4.12 - Ensure a log metric filter and alarm exist for changes to network gateways
- [x] 4.13 - Ensure a log metric filter and alarm exist for route table changes
- [x] 4.14 - Ensure a log metric filter and alarm exist for VPC changes
- [x] 4.15 - Ensure a log metric filter and alarm exists for AWS Organizations changes

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/terraform/aws/missing_log_metric_filter_or_alarm_for_iam_policy_changes/metadata.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "id": "eaaba502-2f94-411a-a3c2-83d63cc1776d",
+  "queryName": "Missing Log Metric Filter Or Alarm For IAM Policy Changes",
+  "severity": "MEDIUM",
```
Comment (JoaoDuarteGomes) [inline assets/queries/terraform/aws/missing_log_metric_filter_or_alarm_for_iam_policy_changes/metadata.json:4]: This severity should be upped to High.

**Review by JoaoDuarteGomes — CHANGES_REQUESTED** [review]
At least the Medium->High changes need to be made. The Medium->Low changes are recommended.

**Review by JoaoDuarteGomes — DISMISSED** [review]
Severities and categories LGTM.

**Review by joaoReigota1 — APPROVED** [review]
LCTM

**Pull request merged**